sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java

```diff
@@ -394,6 +396,17 @@ public CosmosClientTelemetryConfig enableTransportLevelTracing() {
         this.isTransportLevelTracingEnabled = true;
         return this;
     }
+    
+    /**
+     * Enables printing query in db.statement attribute. By default, query is not printed.
+     * Users have to option to enable it, but has to beware customer data may be shown.
+     * It's the user's responsiblity to sanitize the query if necessary.
+     * @return current CosmosClientTelemetryConfig
+     */
+    public CosmosClientTelemetryConfig showQueryStatement() {
```

Comment:
Does the case of turning on/off the showing of queries in traces on the same client (not creating new one) needs to be supported? I.e. dynamic change on/off? Looks like here one can set it on but can't flip it to off.

Reply:
i was thinking of having too flags also, but that would be too much and decided to move the logic to the user custom tracer implementation.  We can discuss more on our call.